### ISSUE TITLE
Implement DIMER Chronos-2 finetuner with LoRA/full training and export

### ISSUE THREAD

**Issue by kurtvalcorza** [kurtvalcorza/chronos-forecasting-finetuner#1]
## Goal
Implement the deployable DIMER finetuner for `amazon/chronos-2` using the official `chronos-forecasting` stack.

DIMER-facing root contract:

```text
Dockerfile
train.py
requirements.txt
README.md
dimer-pipeline.json
```

## Upstream training path
Use `Chronos2Pipeline.fit()` rather than reimplementing Chronos training internals. Current upstream supports:
- `finetune_mode="full" | "lora"`
- validation inputs
- prediction/context length
- learning rate
- number of steps
- batch size
- LoRA config
- output directory
- trainer kwargs/callbacks

Reference: https://github.com/amazon-science/chronos-forecasting

## DIMER runtime contract
- Root entrypoint: `train.py`.
- Read dataset from `DIMER_DATASET_DIR`.
- Write structured training result to `DIMER_RESULT_PATH`.
- Read preprocessing selections from `DIMER_PREPROCESSING_ARGS_JSON`.
- Read fine-tuning selections from `DIMER_HYPERPARAMETERS_JSON`.
- Read the configured Base Model from DIMER pipeline metadata; **do not declare `model_id` in `dimer-pipeline.json`**.
- Bake/use `DIMER_TASK_TYPE=time_series_forecasting` as the safe fallback for Custom/Other.
- Fail fast on an unsupported base model rather than silently training another model.

## Base model
Initial supported model: `amazon/chronos-2` (Apache-2.0).

Pin dependencies and record the actual resolved model revision/commit where possible for provenance.

## Data ingestion and splitting
Consume the canonical long-format contract defined by the validator:
- timestamp
- numeric target
- optional item_id
- optional covariates

Support explicit chronological `train/val/test` files. If only a single dataset is supplied, create deterministic chronological holdouts. **Never random-split time-series observations.**

Use official Chronos preprocessing helpers where appropriate, particularly for long-format DataFrames and covariates.

## `dimer-pipeline.json`
Manifest keys must map 1:1 to code. Initial preprocessing schema should cover:
- `timestamp_column`
- `target_column`
- `id_column`
- `frequency`
- `validation_fraction`
- `test_fraction`
- `missing_value_strategy`
- `context_length`
- `prediction_length`

Initial `modelFinetuning` schema should cover:
- `finetune_mode` (`lora`, `full`; default `lora`)
- `learning_rate`
- `num_steps`
- `batch_size`
- `seed`
- `lora_r`
- `lora_alpha`
- `lora_dropout`

Only expose parameters the runtime actually consumes. Validate conditional parameters, e.g. LoRA settings should not affect full fine-tuning.

## Training behavior
- Set deterministic seeds across Python/NumPy/PyTorch where applicable.
- Prefer single-GPU execution by default to align with DIMER resource assumptions.
- Detect CUDA availability and report device/runtime metadata.
- Ensure errors from OOM, model download, malformed data and unsupported configuration become useful structured results.
- Do not silently fall back from LoRA to full fine-tuning if PEFT is missing; treat missing required dependency/config as an actionable failure.

## Evaluation
Score the chronological holdout and explicit test split when supplied. Report at least:
- MAE
- RMSE
- WAPE or sMAPE with explicit zero handling
- probabilistic/quantile loss where available
- per-series and aggregate summaries

Include a simple last-value and/or seasonal-naive baseline where feasible.

Report both pre-fine-tune (zero-shot) and post-fine-tune metrics when runtime cost is acceptable; label them unambiguously.

## Artifact export
Produce a self-contained reloadable artifact directory/archive containing:
- fine-tuned full checkpoint or PEFT adapter
- base model id + revision
- model/config files required for reload
- dependency/runtime versions
- schema/preprocessing metadata
- timestamp/id/target/covariate configuration
- context/prediction lengths
- training hyperparameters and seed
- evaluation metrics
- provenance/checksums
- minimal inference script/README

Test **train → save → reload → predict** using the actual Chronos stack.

## Container/runtime
- Root-buildable Dockerfile.
- GPU-compatible PyTorch image/dependencies appropriate for DIMER.
- Keep image reasonably lean; model weights should not be baked into the image unless platform requirements demand it.
- Use cache locations compatible with ephemeral containers.
- Avoid writing outside expected work/output locations.

## Tests
Include:
- manifest↔runtime key parity test
- preprocessing/split tests
- LoRA config mapping test
- full-vs-LoRA behavior tests with mocks where needed
- evaluation metric tests
- structured result/error tests
- artifact metadata tests
- lightweight model-interface tests
- optional real-stack smoke test gated from normal CI
- Docker build/entrypoint smoke

Canonical DIMER test data: deterministic small subset of UCI ElectricityLoadDiagrams20112014 (CC BY 4.0).

## Acceptance criteria
- [ ] Root `train.py`, `Dockerfile`, `requirements.txt`, `README.md`, `dimer-pipeline.json` present.
- [ ] Docker builds from repository root.
- [ ] `DIMER_PREPROCESSING_ARGS_JSON` and `DIMER_HYPERPARAMETERS_JSON` consumed exactly as declared.
- [ ] No `model_id` field in manifest.
- [ ] Base Model is validated as `amazon/chronos-2` for v1.
- [ ] Custom/Other task fallback implemented.
- [ ] Chronological leakage-safe splitting.
- [ ] Official Chronos `fit()` path used.
- [ ] LoRA and full modes correctly implemented.
- [ ] Explicit test split genuinely scored.
- [ ] Zero-shot vs fine-tuned metrics clearly distinguished if both reported.
- [ ] Reloadable artifact produced.
- [ ] Real-stack save/reload/predict smoke passes.
- [ ] CI passes.
- [ ] DIMER on-platform smoke training passes before enablement.

Related umbrella spec: `kurtvalcorza/chronos-forecasting-pipeline` issue #1.
